lib/elixir_bench_web/controllers/webhooks_controller.ex

```diff
@@ -0,0 +1,62 @@
+defmodule ElixirBenchWeb.WebHooks do
+  use ElixirBenchWeb, :controller
+
+  alias ElixirBench.{Benchmarks, Repos, Benchmarks.Job}
+
+  action_fallback(ElixirBenchWeb.FallbackController)
+  @accepted_events ["ping", "push", "pull_request"]
+
+  def handle(conn, params) do
+    with {:ok, payload} <- check_payload_params(params),
+         {:ok, event} <- validate_event_headers(conn) do
+      process_event(event, conn, payload)
+    end
+  end
+
+  defp process_event("ping", conn, _payload) do
+    conn
+    |> json(%{message: "pong"})
+  end
+
+  defp process_event("pull_request", conn, payload) do
+    # data is fetched from sender's repository
+    slug = payload["pull_request"]["head"]["repo"]["full_name"]
+    branch_name = payload["pull_request"]["head"]["ref"]
+    commit_sha = payload["pull_request"]["head"]["sha"]
```

Comment:
you might be able to use helper functions such as `get_in` to make the access nicer here